FirebaseMessaging/Sources/Token/FIRMessagingTokenInfo.m

```diff
@@ -162,15 +158,19 @@ - (nullable instancetype)initWithCoder:(NSCoder *)aDecoder {
 
   FIRMessagingAPNSInfo *APNSInfo = nil;
   if (rawAPNSInfo) {
-    NSError *error;
-    APNSInfo = [GULSecureCoding
-        unarchivedObjectOfClasses:[NSSet setWithObjects:FIRMessagingAPNSInfo.class, nil]
-                         fromData:rawAPNSInfo
-                            error:&error];
-    if (error) {
-      FIRMessagingLoggerDebug(
-          kFIRMessagingMessageCodeTokenInfoBadAPNSInfo,
-          @"Could not parse raw APNS Info while parsing unarchived token info: %@", error);
+    // TODO(chliangGoogle: Use the new API and secureCoding protocol.
+    @try {
+      [NSKeyedUnarchiver setClass:[FIRMessagingAPNSInfo class]
```

Comment:
If we override the class globally then:
- we can still use `GULSecureCoding` since the override will affect it as well
- if the FCM version is used with IID then `FIRInstanceIDAPNSInfo` archives will be unarchived to `FIRMessagingAPNSInfo` in IID as well. It will probably not cause any issue as long as the classes are identical due to Objective-C dynamic nature, but it looks a bit risky and not clean.

I think we should modify it either:
1. override the class only for an `NSKeyedUnarchiver` instance, not globally. It's a safer option.
2. if we are fine with the global override, then the revert is not required. Just appending the global `[NSKeyedUnarchiver setClass:...]` will be enough.

Reply:
I suggest we made the change after I/O breaking change because Messaging already had significant amount of code change and at this point we want to ensure no unnecessary changes in so we can fully tested all the complicated scenarios with IID and Messaging.